test/functional/wallet_resendwallettransactions.py

```diff
@@ -69,9 +54,17 @@ def run_test(self):
         node.p2ps[1].sync_with_ping()
         assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
 
-        self.log.info("Transaction should be rebroadcast after 30 minutes")
-        # Use mocktime and give an extra 5 minutes to be sure.
-        rebroadcast_time = int(time.time()) + 41 * 60
+        # Bump time 12 hours to check the transaction is still not
+        # rebroadcast
+        rebroadcast_time = int(time.time()) + 12 * 60 * 60
+        node.setmocktime(rebroadcast_time)
+        node.p2ps[1].sync_with_ping()
+        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
+
+        self.log.info("Bump time & check that transaction is rebroadcast")
+        # Transaction should be rebroadcast approximately 24 hours in the future,
+        # but can range from 12-36. So bump 36 hours to be sure.
+        rebroadcast_time = int(time.time()) + 36 * 60 * 60
```

Comment:
Suggestion as per https://github.com/bitcoin/bitcoin/pull/18038#discussion_r415750840 with only one call to `time.time()` and two info logging statements.

```python
        # Check transaction is rebroadcast in 12-36 hour time interval.
        time_now = int(time.time())

        self.log.info("Bump time 12 hours from now, check transaction is not yet rebroadcast")
        rebroadcast_time = time_now  + (12 * 60 * 60)
        node.setmocktime(rebroadcast_time)
        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)

        self.log.info("Bump time 36 hours from now, check transaction is rebroadcast")
        rebroadcast_time = time_now + (36 * 60 * 60)
        node.setmocktime(rebroadcast_time)
        wait_until(lambda: node.p2ps[1].tx_invs_received[txid] >= 1, lock=mininode_lock)
```

Reply:
so, I took a closer look at this part of the test, and as @jnewbery pointed out [in that thread](https://github.com/bitcoin/bitcoin/pull/18038#discussion_r417408126), the test wasn't actually checking anything (not just the additional check after 12 hour bump, but the existing one as well).

I've updated so there's just one "transaction should not be rebroadcast" check, and that it occurs after we've bumped almost 12 hours into the future.